src/Workspaces/CSharp/Portable/CodeGeneration/CSharpCodeGenerationService.cs

```diff
@@ -482,6 +483,16 @@ protected override TDeclarationNode AddMembers<TDeclarationNode>(TDeclarationNod
             {
                 return (accessorDeclaration.Body == null) ? destinationMember : Cast<TDeclarationNode>(accessorDeclaration.AddBodyStatements(StatementGenerator.GenerateStatements(statements).ToArray()));
             }
+            else if (destinationMember is CompilationUnitSyntax compilationUnit)
+            {
+                var insertionIndex = compilationUnit.Members.LastIndexOf(memberDeclaration => memberDeclaration.IsKind(SyntaxKind.GlobalStatement)) + 1;
+                return Cast<TDeclarationNode>(compilationUnit.WithMembers(compilationUnit.Members.InsertRange(insertionIndex, StatementGenerator.GenerateStatements(statements).Select(generated => SyntaxFactory.GlobalStatement(generated)).ToArray())));
+            }
+            else if (destinationMember is StatementSyntax statement && statement.IsParentKind(SyntaxKind.GlobalStatement))
```

Comment:
this block i don't quite understand.  what's the case where the destinatino *member* is a statement itself?  who is adding statements *to* a statement itself?

Reply:
Top-level statements of a script are not part of a statement container. In this case, we have to create the container and move the original statement into it along with the new statements.

Reply:
can you give a concrete example of the type of statement we are adding, and the statement it is being added to?  it's not clear to me.

Reply:
This behavior supports the following test:

https://github.com/dotnet/roslyn/blob/6573bd148ffb33ea6a526aef2f2194e8ea0e1059/src/EditorFeatures/CSharpTest/CodeActions/ExtractMethod/ExtractLocalFunctionTests.cs#L4790-L4808

If we didn't wrap the generated code in a block, the generated function would be a _method_ declaration as opposed to a _local function_ declaration.

Reply:
would it make sense to disable extract-local-function here instead?  this really doesn't seem like a desirable change to me.  Plus, the addition of the block seems like it can actually screw things up.  For example, what if we had `var v = [|1 + 1|];`.  Would we make a block that only contained the variable and the local-func, causing references to 'v' to not work?

Reply:
It would make sense to disable, but that was a bit more complicated and only impacts script scenarios. I'll file a follow-up issue for it.